dotnet/src/SemanticKernel.Core/Kernel.cs

```diff
@@ -209,9 +209,16 @@ public async Task<KernelResult> RunAsync(ContextVariables variables, Cancellatio
 
                 context = functionResult.Context;
 
+                var functionInvokedArgs = this.OnFunctionInvoked(functionDetails, functionResult);
+
+                if (functionInvokedArgs is not null)
+                {
+                    // All changes to the SKContext by invoked handlers may reflect in the original function result
+                    functionResult = new FunctionResult(functionDetails.Name, functionDetails.PluginName, functionInvokedArgs.SKContext, functionInvokedArgs.SKContext.Result);
```

Comment:
In case of passing `functionInvokedArgs.SKContext.Result` as last parameter of `FunctionResult`, we will override the actual result from function with result from SKContext, which shouldn't be the case. I will put more description in next comment in unit tests.

Reply:
This might not be the best use case for it but a new design needs to be done to allow changes to the result.